Vagrantfile: install go-bindata

aalemayhu opened the pull request:
Closes: #859 (vagrant: install go-bindata by default)
Signed-off-by: Alexander Alemayhu <alexander@alemayhu.com>

aalemayhu committed:
Vagrantfile: install go-bindata

This is useful to have by default so we can detect uncommitted BPF
changes in the VM.  After this commit missing updates to go-bindata will
cause the CI build to fail.

Closes: #859 (vagrant: install go-bindata by default)
Signed-off-by: Alexander Alemayhu <alexander@alemayhu.com>

aanm commented on Vagrantfile:
```diff
@@ -19,6 +19,7 @@ echo 'cd ~/go/src/github.com/cilium/cilium' >> /home/vagrant/.bashrc
 SCRIPT
 
 $build = <<SCRIPT
+go get -u github.com/jteeuwen/go-bindata/...
```
Be aware this will be run every time someone reloads the VM. I think this can be moved to `$bootstrap` since that repository hasn't received any commits since 2015. And there's no need for the `-u` in `$bootstrap`.

aalemayhu replied:
Moving it to bootstrap[0] does not work
```bash
==> cilium-master: /tmp/vagrant-shell: line 5: go: command not found
The SSH command responded with a non-zero exit status. Vagrant
assumes that this means the command failed. The output for this command
should be in the log above. Please read the output to determine what
went wrong.
```

What am I missing?

[0]:
```diff
diff --git a/Vagrantfile b/Vagrantfile
index ab5b40c198a6..a6657cd93479 100644
--- a/Vagrantfile
+++ b/Vagrantfile
@@ -16,10 +16,10 @@ chown -R vagrant:vagrant /home/vagrant/go
 sudo apt-get -y update || true
 sudo apt-get -y install socat curl jq realpath pv tmux
 echo 'cd ~/go/src/github.com/cilium/cilium' >> /home/vagrant/.bashrc
+go get github.com/jteeuwen/go-bindata/...
 SCRIPT
 
 $build = <<SCRIPT
-go get -u github.com/jteeuwen/go-bindata/...
 ~/go/src/github.com/cilium/cilium/common/build.sh
 rm -fr ~/go/bin/cilium*
 SCRIPT
```

aanm replied:
use `sudo -E env PATH="${PATH}" go get..` or use the `/usr/local/go/bin/go get...`. See which permissions both commands put in the `go-bindata` binary

aalemayhu committed:
cilium: fix outdated bindata

This would not normally be required for non bpf related changes, but the
bindata in master is not up to date. This commit is required for the
build to succeed.

Fixes: bbac42919801 (cilium: bindata, 2017-06-02)
Signed-off-by: Alexander Alemayhu <alexander@alemayhu.com>
